extensions/ipynb/src/notebookSerializer.ts

```diff
@@ -70,33 +91,67 @@ export class NotebookSerializer implements vscode.NotebookSerializer {
 		return data;
 	}
 
-	public serializeNotebook(data: vscode.NotebookData, _token: vscode.CancellationToken): Uint8Array {
-		return new TextEncoder().encode(this.serializeNotebookToString(data));
-	}
-
-	public serializeNotebookToString(data: vscode.NotebookData): string {
-		const notebookContent = getNotebookMetadata(data);
-		// use the preferred language from document metadata or the first cell language as the notebook preferred cell language
-		const preferredCellLanguage = notebookContent.metadata?.language_info?.name ?? data.cells.find(cell => cell.kind === vscode.NotebookCellKind.Code)?.languageId;
+	public async serializeNotebook(data: vscode.NotebookData, _token: vscode.CancellationToken): Promise<Uint8Array> {
+		if (this.disposed) {
+			return new Uint8Array(0);
+		}
 
-		notebookContent.cells = data.cells
-			.map(cell => createJupyterCellFromNotebookCell(cell, preferredCellLanguage))
-			.map(pruneCell);
+		if (this.experimentalSave) {
+			return this.serializeViaWorker2(data);
```

Comment:
@nojaf Thanks for the great work, 
I've made one minor change (looks like a lot though),
basically I'm sending the entire `vscode.NotebookData` to the worker, and processing everything there.
That object can be safely serialized and sent to the worker

This way, all of the cup intensive work is done in worker, including constructing the JSON, serialization to bytes and the sorting of object keys.

& the other change, it to use a single worker as opposed to starting one each time,
this saves a lot more time

Reply:
Hi @DonJayamanne! 

Thank you so much for refining this PR! It’s such a significant improvement over my original code, and I genuinely appreciate the effort you put into it—this would have taken me quite a while to figure out on my own.

Everything is still functioning perfectly on my end, and I'm excited about moving forward with this!

Thanks again! If there’s anything else I can do to help, just give me a shout!